### PR TITLE
feat: group change event mode membership approval mode

### PR DESCRIPTION
**Pull request by viniciusgdr**

### CODE REVIEW

**Comment by viniciusgdr** [comment]
Why wasn't it approved? @tulir

File: types/group.go [hunk 1]
```diff
@@ -40,6 +41,10 @@ type GroupInfo struct {
 	MemberAddMode GroupMemberAddMode
 }
 
+type GroupMembershipApprovalMode struct {
+	IsApprovalRequired bool
```
Comment (tulir) [inline types/group.go:45]:
```suggestion
	IsJoinApprovalRequired bool
```

File: group.go [hunk 2]
```diff
@@ -810,6 +815,10 @@ func (cli *Client) parseGroupChange(node *waBinary.Node) (*events.GroupInfo, err
 			if err != nil {
 				return nil, fmt.Errorf("failed to parse group unlink node in group change: %w", err)
 			}
+		case "membership_approval_mode":
+			evt.MembershipApprovalMode = &types.GroupMembershipApprovalMode{
+				IsApprovalRequired: true,
```
Comment (tulir) [inline group.go:820]:
```suggestion
				IsJoinApprovalRequired: true,
```

File: group.go [hunk 3]
```diff
@@ -640,7 +644,8 @@ func (cli *Client) parseGroupNode(groupNode *waBinary.Node) (*types.GroupInfo, e
 			group.DefaultMembershipApprovalMode = childAG.OptionalString("default_membership_approval_mode")
 		case "incognito":
 			group.IsIncognito = true
-		// TODO: membership_approval_mode
+		case "membership_approval_mode":
+			group.IsApprovalRequired = true
```
Comment (tulir) [inline group.go:648]:
```suggestion
			group.IsJoinApprovalRequired = true
```

**Review by tulir — APPROVED** [review]

**Pull request merged**